packages/ipfs-core/src/components/dag/export.js

```diff
@@ -80,52 +81,30 @@ export function createExport ({ repo, preload, codecs }) {
 }
 
 /**
+ * @template T
  * @param {IPFSRepo} repo
- * @param {AbortOptions} options
- * @param {CID} cid
  * @param {BlockWriter} writer
- * @param {import('ipfs-core-utils/multicodecs').Multicodecs} codecs
- * @param {Set<string>} seen
- * @returns {Promise<void>}
- */
-async function traverseWrite (repo, options, cid, writer, codecs, seen = new Set()) {
-  const b58Cid = cid.toString(base58btc)
-  if (seen.has(b58Cid)) {
-    return
-  }
-
-  const block = await getBlock(repo, options, cid, codecs)
-
-  log(`Adding block ${cid} to car`)
-  await writer.put(block)
-  seen.add(b58Cid)
-
-  // recursive traversal of all links
-  for (const link of block.links) {
-    await traverseWrite(repo, options, link, writer, codecs, seen)
-  }
-}
-
-/**
- * @param {IPFSRepo} repo
  * @param {AbortOptions} options
- * @param {CID} cid
  * @param {import('ipfs-core-utils/multicodecs').Multicodecs} codecs
- * @returns {Promise<{cid:CID, bytes:Uint8Array, links:CID[]}>}
+ * @returns {(cid:CID)=>Promise<Block<T>>}
  */
-async function getBlock (repo, options, cid, codecs) {
-  const bytes = await repo.blocks.get(cid, options)
-
-  /** @type {CID[]} */
-  let links = []
-  const codec = await codecs.getCodec(cid.code)
+function makeLoader (repo, writer, options, codecs) {
+  return async (cid) => {
+    const bytes = await repo.blocks.get(cid, options)
+    const codec = await codecs.getCodec(cid.code)
+
+    let block
+    if (codec) {
+      block = createUnsafe({ bytes, cid, codec })
+    } else if (NO_LINKS_CODECS.includes(cid.code)) {
+      // an opaque block with that will yield no links(), which is all walk() cares about
+      block = createUnsafe({ bytes, cid, value: null })
```

Comment:
This is a bit of a hack; maybe OK for now.

I'd like to do something like go-ipld-prime's [`SkipMe`](https://pkg.go.dev/github.com/ipld/go-ipld-prime/traversal#SkipMe), but the equivalent isn't very JS—it's returned as an error from the load function and the traversal engine checks if the error is a `SkipMe` and then ignores that branch. I don't really want to be throwing errors that are not errors.

We _could_ return `null` in the case of a `SkipMe`, that would be more JSish and leave us with a return type of `Promise<Block<T>|null>` and give us the ability to throw proper errors still.